pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -902,3 +903,38 @@ func ValidateValidatingAdmissionPolicyUpdate(newC, oldC *admissionregistration.V
 func ValidateValidatingAdmissionPolicyBindingUpdate(newC, oldC *admissionregistration.ValidatingAdmissionPolicyBinding) field.ErrorList {
 	return validateValidatingAdmissionPolicyBinding(newC)
 }
+
+func validateValidatingAdmissionPolicyStatus(status *admissionregistration.ValidatingAdmissionPolicyStatus, fldPath *field.Path) field.ErrorList {
+	return validateTypeChecking(status.TypeChecking, fldPath.Child("typeChecking"))
+}
+
+func validateTypeChecking(typeChecking *admissionregistration.TypeChecking, fldPath *field.Path) field.ErrorList {
+	if typeChecking == nil {
+		return nil
+	}
+	return validateExpressionWarnings(typeChecking.ExpressionWarnings, fldPath.Child("expressionWarnings"))
+}
+
+func validateExpressionWarnings(expressionWarnings []admissionregistration.ExpressionWarning, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+	for i, warning := range expressionWarnings {
+		allErrors = append(allErrors, validateExpressionWarning(&warning, fldPath.Index(i))...)
+	}
+	return allErrors
+}
+
+func validateExpressionWarning(expressionWarning *admissionregistration.ExpressionWarning, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+	if expressionWarning.Expression == "" {
+		allErrors = append(allErrors, field.Required(fldPath.Child("expression"), "expression must be non-empty if specified"))
```

Comment:
if we get here, we're validating a specified warning item so can drop the "if specified" qualifiers. the field name is also already included in the field path for the error, so can drop that

```suggestion
		allErrors = append(allErrors, field.Required(fldPath.Child("expression"), "must be non-empty"))
```

Reply:
Good point. Removed all.